### PR TITLE
fix(mongodb): mongodb storage mget must return undefined when key not exist

### LINKED ISSUE #6441: MongoStore does not work as Cache for embeddings
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```typescript
import {AzureOpenAIEmbeddings} from "@langchain/openai";
import {CacheBackedEmbeddings} from "langchain/embeddings/cache_backed";
import {InMemoryStore} from "@langchain/core/stores";
import {MongoClient} from "mongodb";
import {MongoDBStore} from "@langchain/mongodb";
import {LocalFileStore} from "langchain/storage/file_system";


(async () => {
    const underlyingModel = new AzureOpenAIEmbeddings({
        modelName: 'text-embedding-3-large',
        azureOpenAIApiInstanceName: '',
        apiKey: '',
        azureOpenAIApiDeploymentName: '',
        azureOpenAIApiVersion: '2023-05-15',
    });


    const client = new MongoClient('mongodb://localhost:27017');
    const collection = client.db("test_db").collection("embeddings_cache");
    const mongoStore = new MongoDBStore({
        collection,
    });

    const inMemoryStore = new InMemoryStore();
    const filestore = new LocalFileStore({
        rootPath: '/tmp',
    })

    const store = mongoStore;

    await store.mset([
        [
            "1",
            new TextEncoder().encode(JSON.stringify("foo")),
        ]
    ])

    const result = await store.mget(["1"]);
    console.log('direct usage test', result);

    const model = CacheBackedEmbeddings.fromBytesStore(
        underlyingModel,
        store,
        {
            namespace: underlyingModel.modelName,
        }
    );

    const embeddings = await model.embedDocuments(["test"]);
    console.log('embeddings result', embeddings);
})();
```

### Error Message and Stack Trace (if applicable)

```
roman.tyshyk@rtyshyk16 rag-typescrypt % ts-node test.ts
direct usage test [ Uint8Array(5) [ 34, 102, 111, 111, 34 ] ]
embeddings result []
```

### Description

Embeddings from underlyingModel are not returned, as well as nothing has been stored in Mongodb.
It works well with other storages, such as `InMemoryStore` and `LocalFileStore`; see the some code example.

### System Info

```
  "dependencies": {
    "@langchain/azure-openai": "^0.0.11",
    "@langchain/community": "^0.2.24",
    "@langchain/mongodb": "^0.0.5",
    "@langchain/openai": "^0.2.6",
    "langchain": "^0.2.12",
    "mongodb": "^6.8.0",
```

### PR DESCRIPTION
**Pull request by rtyshyk**
Fixes #6441 (issue)

`mget` method in `BaseStore` must return values or undefined in the same order as the input keys. 

PR is fixing MongoDB storage implementation.

### CODE REVIEW

File: libs/langchain-mongodb/src/storage.ts [hunk 1]
```diff
@@ -97,13 +97,19 @@ export class MongoDBStore extends BaseStore<string, Uint8Array> {
...
 
     const encoder = new TextEncoder();
-    return retrievedValues.map((value) => {
-      if (!("value" in value)) {
+    const valueMap = new Map(
+      retrievedValues.map((item) => [item[this.primaryKey], item])
+    );
+
+    return keys.map((key) => {
+      const prefixedKey = this._getPrefixedKey(key);
```
Comment (jacoblee93) [inline libs/langchain-mongodb/src/storage.ts:105]: Should just map over `prefixedKeys`?

File: libs/langchain-mongodb/src/storage.ts [hunk 2]
```diff
@@ -97,13 +97,19 @@ export class MongoDBStore extends BaseStore<string, Uint8Array> {
...
+      const value = valueMap.get(prefixedKey);
+
+      if (!value || !("value" in value)) {
```
Comment (jacoblee93) [inline libs/langchain-mongodb/src/storage.ts:108]: `valueMap` contains tuples now, so this would never be true?

**Comment by jacoblee93** [comment]
Sorry for the delay! A few questions

File: libs/langchain-mongodb/src/storage.ts (hunk 1 above)
Comment (rtyshyk) [reply libs/langchain-mongodb/src/storage.ts]: agree, done.

**Comment by rtyshyk** [comment]
> Sorry for the delay! A few questions

not a problem. review updated. please have a look.

**Comment by rtyshyk** [comment]
@jacoblee93 can you please merge it?

**Comment by jacoblee93** [comment]
Sorry about the delay, will ship shortly!

**Review by jacoblee93 — APPROVED** [review]

**Pull request merged**